selvpc/regions.go

```diff
@@ -13,3 +21,51 @@ func expandResellV2Regions(rawRegions *schema.Set) []string {
 
 	return expandedRegions
 }
+
+// expandResellV2CrossRegionOpts converts the provided rawRegions structure to
+// the slice of crossregionsubnets.CrossRegionOpt.
+func expandResellV2CrossRegionOpts(rawRegions *schema.Set) ([]crossregionsubnets.CrossRegionOpt, error) {
+	rawRegionsLen := rawRegions.Len()
+	if rawRegionsLen == 0 {
+		return nil, errors.New("got empty regions")
+	}
+
+	// Pre-allocate memory for expandedCrossRegionOpts slice.
+	expandedCrossRegionOpts := make([]crossregionsubnets.CrossRegionOpt, rawRegionsLen)
+
+	// Iterate over each value in rawRegions and add it to expandedCrossRegionOpts slice.
+	for i, rawRegion := range rawRegions.List() {
+		var region string
+		mapRegion := rawRegion.(map[string]interface{})
+		if value, ok := mapRegion["region"]; ok {
+			region = value.(string)
+		}
+		expandedCrossRegionOpts[i] = crossregionsubnets.CrossRegionOpt{
+			Region: region,
+		}
+	}
+
+	return expandedCrossRegionOpts, nil
+}
+
+// regionsMapsFromSubnetsStructs converts the provided subnets.Subnet to
+// the slice of maps with associated regions correspondingly to the resource's schema.
+func regionsMapsFromSubnetsStructs(subnetsStructs []subnets.Subnet) []map[string]interface{} {
+	associatedRegions := make([]map[string]interface{}, len(subnetsStructs))
+
+	for i, subnet := range subnetsStructs {
+		associatedRegions[i] = make(map[string]interface{})
+
+		if subnet.Region != "" {
```

Comment:
We will have slice with empty maps if subnet.Region == "" because we initialised associatedRegions[i] with map.
So, final slice will look like this:
```
[]map[string]interface{}{
    {"region": "ru-1"},
    {},
    {"region": "ru-3"},
}
```